x-pack/packages/observability/observability_utils/observability_utils_server/es/client/create_observability_es_client.ts

```diff
@@ -109,32 +123,35 @@ export function createObservabilityEsClient({
         });
       });
     },
-    esql<TOutput = unknown, TSearchRequest extends EsqlParameters = EsqlParameters>(
+    esql<
+      TOutput extends EsqlOutput = EsqlOutput,
+      TEsqlOptions extends EsqlOptions | undefined = { asPlainObjects: true }
+    >(
       operationName: string,
-      { parseOutput = true, format = 'json', columnar = false, ...parameters }: TSearchRequest
-    ) {
-      logger.trace(() => `Request (${operationName}):\n${JSON.stringify(parameters, null, 2)}`);
-      return withSpan({ name: operationName, labels: { plugin } }, () => {
-        return client.esql.query(
-          { ...parameters, format, columnar },
-          {
-            querystring: {
-              drop_null_columns: true,
-            },
-          }
-        );
-      })
-        .then((response) => {
-          logger.trace(() => `Response (${operationName}):\n${JSON.stringify(response, null, 2)}`);
+      parameters: EsqlQueryRequest,
+      options?: EsqlOptions
+    ): Promise<InferEsqlResponseOf<TOutput, TEsqlOptions>> {
+      return callWithLogger(operationName, parameters, () => {
+        return client.esql
+          .query(
+            { ...parameters },
+            {
+              querystring: {
+                drop_null_columns: true,
+              },
+            }
+          )
+          .then((response) => {
+            const esqlResponse = response as unknown as EsqlQueryResponse;
 
-          const esqlResponse = response as unknown as ESQLSearchResponse;
+            const shouldParseOutput = options?.asPlainObjects !== false;
+            const finalResponse = shouldParseOutput
```

Comment:
`asPlainObject` doesn't take into account the [`format`](https://www.elastic.co/guide/en/elasticsearch/reference/current/esql-rest.html#esql-rest-format) paramater. In theory, if the function receives anything other than `json` the `esqlResultToPlainObjects` won't work. Or this isn't the case with the Elasticsearch client?

Reply:
it's currently the responsibility of the consumer. I can also throw if `asPlainObjects == true` and we're seeing different values for `format` or `columnar`. I haven't really tried what happens if you're not using `json`, maybe we should just not allow that parameter entirely?

Reply:
`arrow` is probably useful

Reply:
yeah. within Kibana, I don't see reasons for supporting formats other than `arrow` and `json`

Reply:
discussed with @crespocarlos on Slack, I've changed this to a `transform` option which takes `none` (the default), `plain`, and `unflatten`. I've disallowed `format` as an option for now. We can add it back later but we have to do some type gymnastics then wrt the response type which are probably not worth it for now.